test/test_belief_propagation.jl

```diff
@@ -40,13 +41,13 @@ using Test: @test, @testset
   s = siteinds("S=1/2", g)
   χ = 2
   rng = StableRNG(1234)
-  ψ = random_tensornetwork(rng, s; link_space=χ)
+  ψ = random_tensornetwork(rng, ComplexF64, s; link_space=χ)
```

Comment:
A code pattern I use in other tests is to make the entire testset loop over a few element types, for example:
```julia
@testset "belief_propagation (eltype=$elt)" for elt in (Float32, Float64, Complex{Float32}, Complex{Float64})
```
and then use `elt` in the test to construct tensors/tensor networks with different element types. It's also a good way to check that the code preserves the element type properly, by checking that the outputs have element type `elt`, since it is easy to accidentally write code that doesn't preserve single precision (`Float32`), which can cause problems on GPUs.

Reply:
Good idea, I added `eltype` support for the message tensor construction and then added that to the test